handsontable/src/plugins/formulas/formulas.js

```diff
@@ -614,7 +644,35 @@ export class Formulas extends BasePlugin {
       return false;
     }
 
-    return this.engine.getFillRangeData(engineSourceRange, engineTargetRange);
+    const fillRangeData = this.engine.getFillRangeData(engineSourceRange, engineTargetRange);
+    const [sourceStartRow, sourceStartColumn] = [engineSourceRange.start.row, engineSourceRange.start.col];
+    const [sourceEndRow, sourceEndColumn] = [engineSourceRange.end.row, engineSourceRange.end.col];
+    const [populationRowLength, populationColumnLength] = [sourceEndRow - sourceStartRow + 1,
+      sourceEndColumn - sourceStartColumn + 1];
+
+    for (let populatedRowIndex = 0; populatedRowIndex < fillRangeData.length; populatedRowIndex += 1) {
+      for (let populatedColumnIndex = 0; populatedColumnIndex < fillRangeData[populatedRowIndex].length;
+        populatedColumnIndex += 1) {
+        const populatedValue = fillRangeData[populatedRowIndex][populatedColumnIndex];
+        const [sourceRow, sourceColumn] = [populatedRowIndex % populationRowLength,
+          populatedColumnIndex % populationColumnLength];
```

Comment:
Why not use a basic variable assignment? Creating an array and in the same line destructing it. I think it can create more confusion than it's worth.
```suggestion
        const sourceRow = populatedRowIndex % populationRowLength;
        const sourceColumn = populatedColumnIndex % populationColumnLength;
```

Reply:
I did it intentionally to shorten the code, but I don't mind changing it.